parachain/pallets/incentivized-channel/src/outbound/mod.rs

```diff
@@ -116,34 +136,33 @@ pub fn offchain_key(prefix: &[u8], hash: H256) -> Vec<u8> {
 
 
 impl<T: Config> Module<T> {
-	pub fn submit(_: &T::AccountId, target: H160, payload: &[u8]) -> DispatchResult {
-		Nonce::try_mutate(|nonce| -> DispatchResult {
-			*nonce += 1;
-			Self::try_add_message(target, *nonce, payload)?;
-			<Module<T>>::deposit_event(Event::MessageAccepted(*nonce));
-			Ok(())
-		})
-	}
 
-	fn try_add_message(target: H160, nonce: u64, payload: &[u8]) -> DispatchResult {
+	pub fn submit(who: &T::AccountId, target: H160, payload: &[u8]) -> DispatchResult {
 		ensure!(
 			MessageQueue::decode_len().unwrap_or(0) < T::MaxMessagesPerCommit::get(),
 			Error::<T>::QueueSizeLimitReached,
 		);
-
 		ensure!(
 			payload.len() <= T::MaxMessagePayloadSize::get(),
 			Error::<T>::PayloadTooLarge,
 		);
 
-		MessageQueue::append(
-			Message {
-				target,
-				nonce,
-				payload: payload.to_vec(),
-			},
-		);
-		Ok(())
+		let fee = Self::fee();
+		T::FeeCurrency::withdraw(who, fee).map_err(|_| Error::<T>::NoFunds)?;
```

Comment:
where does the fee go? is it burned?

Reply:
assume this works fine too if user is burning eth, then the total burned will be the base message burned + fee burned?

Reply:
yeah, under the covers the amount is withdrawn from the account, and deducted from total issuance, so in effect is burned.